packages/plugins/plugin-token-manager/src/components/OAuthPresetSelector.tsx

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright 2025 DXOS.org
+//
+
+import React, { useEffect, useState } from 'react';
+
+import { createIntent, useIntentDispatcher } from '@dxos/app-framework';
+import { log } from '@dxos/log';
+import { SpaceAction } from '@dxos/plugin-space/types';
+import { type OAuthFlowResult, OAuthProvider } from '@dxos/protocols';
+import { create, type Space } from '@dxos/react-client/echo';
+import { useEdgeClient } from '@dxos/react-edge-client';
+import { Button, DropdownMenu, useTranslation } from '@dxos/react-ui';
+import { AccessTokenType } from '@dxos/schema';
+
+import { TOKEN_MANAGER_PLUGIN } from '../meta';
+
+const OAUTH_PRESETS: OAuthPreset[] = [
+  {
+    label: 'GMail',
+    note: 'Email read access.',
+    source: 'https://gmail.com/',
+    provider: OAuthProvider.GOOGLE,
+    scopes: ['https://www.googleapis.com/auth/gmail.readonly'],
+  },
+];
+
+export const OAuthPresetSelector = ({ space }: { space: Space }) => {
+  const { t } = useTranslation(TOKEN_MANAGER_PLUGIN);
+  const edgeClient = useEdgeClient();
+  const { dispatchPromise: dispatch } = useIntentDispatcher();
+  const [tokenMap] = useState(new Map<string, AccessTokenType>());
+
+  useEffect(() => {
+    const edgeUrl = new URL(edgeClient.baseUrl);
+
+    const listener = (event: MessageEvent) => {
+      if (event.origin === edgeUrl.origin) {
+        const data = event.data as OAuthFlowResult;
+        if (data.success) {
+          const token = tokenMap.get(data.accessTokenId);
+          if (token) {
+            token.token = data.accessToken;
+            dispatch(createIntent(SpaceAction.AddObject, { object: token, target: space })).catch((e) => log.catch(e));
```

Comment:
specify `hidden: true` to ensure the tokens are not added to the root space collection (just landed this option on `main`)